src/main/java/io/neonbee/entity/EntityModelLoader.java

```diff
@@ -187,24 +209,43 @@ Future<Void> loadModel(Path csnFile) {
     }
 
     Future<Void> parseModel(String csnFile, byte[] csnPayload, Map<String, byte[]> associatedModels) {
-        return parseCsnModel(csnPayload)
-                .compose(cdsModel -> CompositeFuture
-                        .all(EntityModelDefinition.resolveEdmxPaths(Path.of(csnFile), cdsModel).stream()
-                                .map(Path::toString).map(path -> {
-                                    // we do not know if the path uses windows / unix path separators, try both!
-                                    return FileSystemHelper.getPathFromMap(associatedModels, path);
-                                }).map(this::parseEdmxModel).collect(Collectors.toList()))
-                        .onSuccess(compositeFuture -> {
-                            buildModelMap(cdsModel, compositeFuture.<ServiceMetadata>list());
-                        }))
-                .mapEmpty();
+        LOGGER.trace("Parse CSN model file {}", csnFile);
+        return parseCsnModel(csnPayload).onComplete(result -> {
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("Parsing CSN model file {} {}", csnFile, result.succeeded() ? "succeeded" : "failed");
+            }
+        }).compose(cdsModel -> {
+            LOGGER.trace("Parse associated models of {}", csnFile);
+            return CompositeFuture.all(EntityModelDefinition.resolveEdmxPaths(Path.of(csnFile), cdsModel).stream()
+                    .map(Path::toString).map(path -> {
+                        // we do not know if the path uses windows / unix path separators, try both!
+                        return FileSystemHelper.getPathFromMap(associatedModels, path);
+                    }).map(this::parseEdmxModel).collect(Collectors.toList())).onComplete(result -> {
+                        if (LOGGER.isTraceEnabled()) {
+                            LOGGER.trace("Parsing associated models of {} {}", csnFile,
+                                    result.succeeded() ? "succeeded" : "failed");
+                        }
+                    }).onSuccess(compositeFuture -> {
+                        buildModelMap(cdsModel, compositeFuture.<ServiceMetadata>list());
```

Comment:
In my opinion, this does not properly handle the error. `buildModelMap` might throw a `RuntimeException`, when building the model map is not possible. In that case I think it would be better to let `parseModel` return a `failedFuture`.

Same with `EntityModelManager.loadModel`.

Reply:
Mhm I agree generally with your statement, that throwing a plain `RuntimeException` is most strange. However I don't think it is required to return a `failedFuture` either... The question is: Is a CDS model w/o a service (in that case `getNamespace` throws the runtime exception) a failure? Well according to the CDS compiler yes, you cannot compile a CDS model w/o a service to EDMX/CSN, however their Java API (`CdsModelBuilder`) allows to create a model w/o a service (I created a test for it). Thus I think the right approach would be: We just ignore models w/o services for now and, like the CDS compiler on empty services, log a warning. That is I think the most "gentle" approach to this, what do you think?

Reply:
> The question is: Is a CDS model w/o a service (in that case getNamespace throws the runtime exception) a failure?

I would say yes, just because it takes much longer for a developer to find out why the endpoint is not showing data if we only write a log (on `TRACE` level which is anyways hard to find - as we write many logs) rather than failing the deployment. I don't think anyone would create such models if it wasn't a mistake.

On the other hand, I'm with you on the argument about the `CdsModelBuilder`, but wouldn't necessarily see that as a requirement that we accept such models.

Reply:
Would you mind taking a look at my current revision of this change, I think the way I now logged it as a WARNING is the right way to go, if you also agree, I would fix the last issues of the voter and merge.